packages/open-collaboration-server/src/auth-endpoints/auth-endpoint.ts

```diff
@@ -0,0 +1,17 @@
+import type { Express } from 'express';
+import { Event } from 'open-collaboration-rpc';
+import { User } from '../types';
+
+export type UserInfo = Omit<User, 'id'>;
+export interface AuthSuccessEvent {
+    token: string, 
+    userInfo: UserInfo
+}
+
+export const AuthEndpoint = Symbol('AuthEndpoint');
+
+export interface AuthEndpoint {
+    shouldActivate(): boolean;
+    onStart(app: Express, hostname: string, port: number): void;
+    onDidSuccessfullyAuthenticate: Event<AuthSuccessEvent>;
```

Comment:
Suggestion: I think we can shorten this name.
```suggestion
    onDidAuthenticate: Event<AuthSuccessEvent>;
```